### PR TITLE
added user-defined callback functions for release and abort requests

### PR DESCRIPTION
**Pull request by stultus**
Just like OnReceiveStore, OnAssociateRequest etc, user should be able to create custom callback functions during the release and abort events.

### CODE REVIEW

File: source/netdicom/applicationentity.py [hunk 1]
```diff
@@ -188,10 +188,14 @@ def run(self):
 
                 # check for release request
                 if self.ACSE.CheckRelease():
+                    if 'OnRelease' in self.AE.__dict__:
```
Comment (cancan101) [inline source/netdicom/applicationentity.py]: Why are you not writing this like `hasattr(self.AE, 'OnRelease')`?

**Comment by cancan101** [comment]
@stultus Bump.

**Comment by stultus** [comment]
Sorry I was afk for a few days.

File: source/netdicom/applicationentity.py (hunk 1 above)
Comment (stultus) [inline source/netdicom/applicationentity.py]: There is no specific reason, I was just following the method already used. 
ie. the OnAssociateResponse is handled this way.  should I rewrite it by using the hasattr method?

Comment (cancan101) [inline source/netdicom/applicationentity.py]: Yes, checking dict seem wrong for a few reasons. I would either use hasattr or just call it and catch the attribute error.

**Comment by cancan101** [comment]
Bump.

**Pull request closed without merging**